### ISSUE TITLE
Rule Proposal: vue/no-deprecated-router-link-tag-prop

### ISSUE THREAD

**Issue by manniL** [vuejs/eslint-plugin-vue#1306]
**Please describe what the rule should do:**

Using the `tag` property for `RouterLink` or `NuxtLink` is usually a bad idea because it removes out-of-the-box accessibility (further: https://github.com/vuejs/vue-router-next/issues/148#issuecomment-605709793). This rule should throw an error if people try to change the tag.

PS: The `tag` prop will be removed in future majors of vue-router.

**What category should the rule belong to?**

[ ] Enforces code style (layout)
[x] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<template>
 <RouterLink tag="div" to="/">Home</RouterLink>
</template>
```

```vue
<template>
 <RouterLink :tag="someVariable" to="/">Home</RouterLink>
</template>
```

```vue
<template>
 <NuxtLink tag="div" to="/">Home</NuxtLink>
</template>
```

**Comment by ota-meshi** [comment]
Thank you for suggesting the rule.
That sounds good to me.

However, rules that disallow deprecated features has `no-deprecated`, so I think a rule name like` no-deprecated-router-link-tag-prop` is good.

I think the documentation for this rule should include a link to the RFC.
https://github.com/vuejs/rfcs/blob/master/active-rfcs/0021-router-link-scoped-slot.md
Also, I think that good to include a hint to use `custom` prop and `v-slot` in the rule warning message.

The rule checks only the RouterLink component by default. Rule options allow users to add more components they want to check. Users can also use this option to check the NuxtLink component.

**Comment by manniL** [comment]
Thanks for the feedback! I've changed the name accordingly 😋
